src/netCLISample/Program.cs

```diff
@@ -8,49 +8,67 @@ namespace netCLISample
 {
     class Program
     {
+        static ILogger<Program> logger;
+
         static int Main(string[] args)
         {
             var services = new ServiceCollection();
             services.AddLogging(builder =>
             {
-                builder.ClearProviders();
-                builder.AddProvider(new MyConsoleLoggerProvider());
+                builder.SetMinimumLevel(LogLevel.Debug);
+                builder.AddConsole();
             });
-            services.ConfigureCLFlow(config =>
+            services.ConfigureCLFlow(mapping =>
             {
-                config.Map<MoveCommand, MoveCommandHandler>()
+                mapping.Map<MoveCommand, MoveCommandHandler>()
                       .Map<HelpCommand, HelpCommandHandler>();
 
             });
 
-            var provider = services.BuildServiceProvider();
+            using( var provider = services.BuildServiceProvider())
+            {
+                logger = provider.GetRequiredService<ILogger<Program>>();
 
-            return CLFlow.Create().UseServiceProvider(provider)
-                                  .UseFallback(ErrorHandling)
-                                  .Run(args);
+                return CLFlow.Create().UseServiceProvider(provider)
+                                      .UseFallback(ErrorHandling)
+                                      .Run(args);
+            }
         }
 
         public static void ErrorHandling(ErrorCode errorCode)
         {
-            Console.WriteLine($"Bla error {errorCode}.");
+            switch (errorCode)
+            {
+                case ErrorCode.RequiredOptionMissing:
+                    logger.LogError("Could not execute command. Some required options were missing.");
+                    break;
+                default:
+                    logger.LogError("Could not execute command due to some general error.");
+                    break;
+            }
+            logger.LogInformation("Try `help --command <command>` to see more details.");
         }
     }
 
-
-
     [Command("move", "Moves files between locations.")]
     public class MoveCommand
     {
-        [Option("force", "Indicates weather or not files should be overwritten.")]
+        [Option("force", "Indicates weather or not files should be overwritten.", IsOptional = true)]
         public bool IsForce { get; set; }
+
+        [Option("from", "Current files locations.")]
+        public string PathFrom { get; set; }
+
+        [Option("to", "Destination path for the files.")]
+        public bool PathTo { get; set; }
     }
 
     public class MoveCommandHandler : CommandHandler<MoveCommand>
     {
         public override int Run(MoveCommand options)
         {
             Console.WriteLine("bla");
-            return 1;
+            return 0;
```

Comment:
What does the return value mean here? Is it used anywhere? Should it always be zero?

Based on the fact that `Executor.Run` method returns the result from command handler:
https://github.com/inausoft/netCLI/blob/c4b0fb5e639039a70135ab8b54611b025ae550d3/src/inausoft.netCLI/Executor.cs#L90
and what we expect to see when running `Executor.Run`:
https://github.com/inausoft/netCLI/blob/c4b0fb5e639039a70135ab8b54611b025ae550d3/src/inausoft.netCLI.Tests/Executor.Tests.cs#L160-L164
I understand that the return value should be interpreted as error code.

Should return type from `Run` method be of type `ErrorCode` then?

What if I decide that my handler returns `10` on some custom action type? Will this be interpreted as `UnspecifiedCommand`?

Reply:
Generally netCLI will forward return code from handler that's all. If the framework will however encounter any problem like parsing it will return code related to ErrorCode enum.